File: src/features/typing/data/go/decode_string.go
package main

func decodeString(s string) string {
	countStack := []int{}
	stringStack := []string{}
	current := ""
	count := 0
	for i := 0; i < len(s); i++ {
		ch := s[i]
		if ch >= '0' && ch <= '9' {
			count = count*10 + int(ch-'0')
		} else if ch == '[' {
			countStack = append(countStack, count)
			stringStack = append(stringStack, current)
			current = ""
			count = 0
		} else if ch == ']' {
			repeat := countStack[len(countStack)-1]
			countStack = countStack[:len(countStack)-1]
			prev := stringStack[len(stringStack)-1]
			stringStack = stringStack[:len(stringStack)-1]
			for j := 0; j < repeat; j++ {
				prev += current
			}
			current = prev
		} else {
			current += string(ch)
		}
	}
	return current
}

File: src/features/typing/data/dart/decode_string.dart
class Solution {
  String decodeString(String s) {
    List<int> countStack = [];
    List<String> stringStack = [];
    String current = "";
    int count = 0;
    for (var i = 0; i < s.length; i++) {
      String ch = s[i];
      int? digit = int.tryParse(ch);
      if (digit != null) {
        count = count * 10 + digit;
      } else if (ch == '[') {
        countStack.add(count);
        stringStack.add(current);
        current = "";
        count = 0;
      } else if (ch == ']') {
        int repeat = countStack.removeLast();
        String prev = stringStack.removeLast();
        for (var j = 0; j < repeat; j++) {
          prev += current;
        }
        current = prev;
      } else {
        current += ch;
      }
    }
    return current;
  }
}
